### PR TITLE
Add support for human-readable time input

### LINKED ISSUE #12: Support human-readable time input
* **I'm submitting a ...**
[ ] bug report
[x] feature request
[ ] question about the decisions made in the repository
[ ] question about how to use this project

* **Summary**

Rather than having to input milliseconds (which can be hard to read at a glance) it would be cool if this package supported https://github.com/zeit/ms. That would allow one to write the limits more human-readable:

```GraphQL
type Mutation {
  createItem(args: CreateItemArgs!): Item @rateLimit(max: 20, window: 300000)
  # compared to
  createItem(args: CreateItemArgs!): Item @rateLimit(max: 20, window: "5m")
}
```

This would be a backwards-compatible change too.

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, personal fork, etc.)

See the [zeit/ms](https://github.com/zeit/ms) readme for more information

### PR DESCRIPTION
**Pull request by mxstbr**
Adds support for human-readable time input as described in #12.

Closes #12

### CODE REVIEW

File: src/lib/field-directive.ts [hunk 1]
```diff
@@ -154,13 +156,19 @@ const createRateLimitDirective = (
             type: new GraphQLList(GraphQLString)
           },
           max: {
-            type: GraphQLInt
+            type: new GraphQLUnionType({
+              name: 'max',
+              types: [GraphQLInt, GraphQLString]
+            })
```
Comment (kirkness) [inline src/lib/field-directive.ts]: `ms` isn't needed on this arg, right? `max` is not a time value

File: src/lib/field-directive.ts [hunk 2]
```diff
@@ -176,8 +184,10 @@ const createRateLimitDirective = (
       field.resolve = async (...args) => {
         const [, resolveArgs, context] = args;
         const contextIdentity = config.identifyContext(context);
-        const window = this.args.window || DEFAULT_WINDOW;
-        const max = this.args.max || DEFAULT_MAX;
+        let window = this.args.window || DEFAULT_WINDOW;
+        let max = this.args.max || DEFAULT_MAX;
+        if (typeof window !== 'number') window = ms(window);
+        if (typeof max !== 'number') max = ms(max);
```
Comment (kirkness) [inline src/lib/field-directive.ts]: ditto

**Comment by kirkness** [comment]
Not sure this is going to be possible the way its implemented. [Union scalars aren't supported in GraphQL](https://github.com/facebook/graphql/issues/215), so will need to re-think the API. Either 2 different arg names, `window` + `windowString` or only support strings? Looks like `ms('1000') === 1000`.

File: src/lib/field-directive.ts (hunk 1 above)
Comment (mxstbr) [reply src/lib/field-directive.ts]: Oh what was I doing, jeeeez 🤦‍♂️

**Comment by mxstbr** [comment]
Hmm damn. I guess only strings would be fine and using `"1000ms"` throughout the docs?

I don't know what I was thinking with the `max` thing, sorry about that!

**Comment by mxstbr** [comment]
Latest commits clean this up!

File: src/lib/field-directive.ts (hunk 1 above)
Comment (kirkness) [reply src/lib/field-directive.ts]: 😂

**Comment by kirkness** [comment]
Done! I'll publish this along with the rest of the updates a little later. Thanks!

**Pull request merged**